packages/plugin-fsapp/src/index.ts

```diff
@@ -0,0 +1,452 @@
+    }
+
+    // Resolve the package name to its filepath
+    packageOrFilePath = require.resolve(packageOrFilePath, {
+      paths: [process.cwd()],
+    });
+  }
+
+  // Use 'bundle-require' to bundle and require the specified file or package
+  const {mod} = await _bundleRequire({
+    filepath: packageOrFilePath,
+    format,
+  });
+
+  // Return the required module
+  return mod;
+}
+
+async function validateEnvFiles(baseDir: string): Promise<string[]> {
+  const pattern = path.join(baseDir, '**', 'env.*.ts');
+  const files = glob.sync(pattern, {ignore: '**/node_modules/**'});
+
+  const results: string[] = [];
+
+  for (const file of files) {
+    const code = await fs.readFile(file, 'utf-8');
+    const ast = parse(code, {
+      sourceType: 'module',
+      plugins: ['typescript'],
+    });
+
+    traverse(ast, {
+      Program(path) {
+        let hasDefaultExport = false;
+        let hasDefineEnvImport = false;
+
+        path.traverse({
+          ExportNamedDeclaration(path) {
+            const declaration = path.node.declaration;
+            if (
+              t.isTypeAlias(declaration) ||
+              t.isInterfaceDeclaration(declaration) ||
+              t.isTSTypeAliasDeclaration(declaration) ||
+              t.isTSInterfaceDeclaration(declaration)
+            ) {
+              // Allows type exports
+              return;
+            }
+
+            throw new Error(
+              chalk.red(
+                dedent`\n\nError: Named exports found in ${file}.
+
+                Flagship Code requires that each "env.*.ts" file contain exactly one default export.
+                Named exports or multiple default exports are not supported and will lead to this error.
+
+                Steps to resolve this issue:
+                1. Open the file where the error occurred.
+
+                2. Verify that the module contains exactly one default export, structured like this:
+
+                       export default defineEnv({
+                         key: 'value',
+                       });
+
+                3. Remove any additional named exports, multiple default exports, or incorrect usage.
+
+                For more details on "env.*.ts" file export requirements, refer to the Flagship Code documentation or contact your support team.
+                `,
+              ),
+            );
+          },
+          ExportDefaultDeclaration() {
+            hasDefaultExport = true;
+          },
+          ImportDeclaration(path) {
+            if (
+              path.node.source.value === '@brandingbrand/code-cli-kit' &&
+              path.node.specifiers.some(
+                specifier =>
+                  t.isImportSpecifier(specifier) &&
+                  t.isIdentifier(specifier.imported) &&
+                  specifier.imported.name === 'defineEnv',
+              )
+            ) {
+              hasDefineEnvImport = true;
+            }
+          },
+        });
+
+        if (!hasDefaultExport) {
```

Comment:
we should likely only throw an error here if the import does not exist that way we know it's supposed to be a runtime env